User Authentication

abruzy opened the pull request:
#### What does this PR do?

- This PR has an implementation of user authentication

#### Description of Task to be completed?

- implementation of user authentication

#### How should this be manually tested?

- start the server by running rails s

#### Any background context you want to provide?

- Nil

closes #7

abruzy committed:
convert rails only API app to normal rails app

abruzy committed:
refactor: revert some changes and added tailwind config

abruzy committed:
refactor: revert some changes

abruzy committed:
refactor: rename some files

abruzy committed:
feat: user authentication

abruzy committed:
feat: user authentication

robocopkaka commented:
Looks like you have commits from a different PR in this. Could you fix this? Doesn't look like you pulled master down before creating this branch

abruzy committed:
feat: user authentication

robocopkaka commented on app/controllers/homes_controller.rb:
```diff
@@ -1,4 +1,6 @@
 class HomesController < ApplicationController
+  before_action :require_login
   def index
+    binding.pry
```
Please remove this

robocopkaka commented on app/controllers/sessions_controller.rb:
```diff
@@ -0,0 +1,29 @@
+    @user = authenticate(params)
+    
+    sign_in(@user) do |status|
+      binding.pry
```
Also this

robocopkaka commented on app/controllers/sessions_controller.rb:
```diff
@@ -0,0 +1,29 @@
+  private
+  
+  def doctor_params
+    params.require(:doctor).permit(
```
Could you change this. Don't think sessions exposes `doctor`

robocopkaka commented on app/models/user.rb:
```diff
@@ -19,6 +20,12 @@ class User < ApplicationRecord
+  #enum
+  enum gender: {
+    male: 0,
+    female: 1,
```
I think there's a 3rd enum type that's missing. Please add it

robocopkaka commented on app/controllers/patients_controller.rb:
```diff
@@ -0,0 +1,30 @@
+class PatientsController < Clearance::UsersController
```
Forgot to mention that we won't need a controller for patients. Please change this to `admins`. We'd only keep a model for patients

robocopkaka reviewed (commented):
Please add feature tests for the features you worked on that can be written using capybara

abruzy committed:
chore: test added and refactor controllers, model and routes

robocopkaka committed:
fix bugs
  - remove calls to `password_confirmation`
  - downcase emails in factories
  - add doctors directory to views folder
  - add link for signing up new doctors to the sessions page
  - add routes for signing up doctors

abruzy committed:
feat: styled sign-in page

abruzy committed:
feat: styled sign-in page

robocopkaka reviewed (approved):
Approved
